src/main/java/io/wispforest/owo/mixin/recipe/RecipeManagerMixin.java

```diff
@@ -0,0 +1,67 @@
+package io.wispforest.owo.mixin.recipe;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+import io.wispforest.owo.util.RecipeSpecificRemainderStorage;
+import net.minecraft.item.Item;
+import net.minecraft.recipe.Recipe;
+import net.minecraft.recipe.RecipeManager;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.JsonHelper;
+import net.minecraft.util.registry.Registry;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import java.util.Map;
+import java.util.Optional;
+
+@Mixin(RecipeManager.class)
+public abstract class RecipeManagerMixin {
+
+    @Inject(method = "deserialize", at = @At(value = "RETURN"))
+    private static void deserializeRecipeSpecificRemainders(Identifier id, JsonObject json, CallbackInfoReturnable<Recipe<?>> cir){
+        if(json.has("owo:remainders")){
+            JsonObject remainders = json.getAsJsonObject("owo:remainders");
+
+            for(Map.Entry<String, JsonElement> entry : remainders.entrySet()){
+                Identifier item_id = Identifier.tryParse(entry.getKey());
+
+                if(item_id == null) {
+                    throw new JsonSyntaxException("OwO Remainders: A recipe Remainder has invalid item Identifier");
+                }
+
+                Optional<Item> item = Registry.ITEM.getOrEmpty(item_id);
```

Comment:
`JsonHelper.asItem()`

Reply:
Not possible with how the API makes remainders: 
`"owo:remainders": {
    "minecraft:sand": {
      "item": "minecraft:sand"
    }
  },`

Reply:
Just slapped it into a JsonPrimitive and ran it into such